### PR TITLE
Strip spacing icl

### PR DESCRIPTION
**Pull request by bmosaicml**
# What does this PR do?

Remove all spaces from the data points in ICL jsonls. We will allow the data loader to then add space back in as requested by the delimiters/required by the tokenizer!

Confirm that Lambada/hellaswag/winograd scores are unchanged for mpt7b instruct!

Before:

| language_understanding   | lambada_openai                   |                                     | 0.69086     |                 0 | mosaicml/mpt-7b-instruct                      |
| language_understanding   | hellaswag                        |                                     | 0.769966    |                 0 | mosaicml/mpt-7b-instruct                      |
| language_understanding   | winograd                         |                                     | 0.846154    |                 0 | mosaicml/mpt-7b-instruct                      |

After

| language_understanding   | lambada_openai                   |                                     |  0.69086   |                 0 | mosaicml/mpt-7b-instruct |
| language_understanding   | hellaswag                        |                                     |  0.769966  |                 0 | mosaicml/mpt-7b-instruct |
| language_understanding   | winograd                         |                                     |  0.846154  |                 0 | mosaicml/mpt-7b-instruct |

# What issue(s) does this change relate to?

# Before submitting
- [ ] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [ ] Is this change a documentation change or typo fix? If so, skip the rest of this checklist.
- [ ] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [ ] Did you update any related docs and document your change?
- [ ] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [ ] Did you run the tests locally to make sure they pass?
- [ ] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

### CODE REVIEW

File: composer/datasets/in_context_learning_evaluation.py [hunk 1]
```diff
@@ -531,7 +540,7 @@ def prep_examples(self, num_fewshot: int, prompt_string: str, example_delimiter:
                         'choices'], self.samples[fewshot_idx]['gold']
                     if len(preamble) > 0:
                         query = f'{example_delimiter}{query}'
-                    preamble += f'{query}{continuation_delimiter}{choices[gold_idx]}'
+                    preamble += f'{query}{continuation_delimiter}{choices[gold_idx]}'  # pyright: ignore [reportGeneralTypeIssues]
```
Comment (dakinggg) [inline composer/datasets/in_context_learning_evaluation.py:543]: why does this need a new type ignore? i don't think it should, so might indicate a bug?

File: composer/datasets/in_context_learning_evaluation.py [hunk 2]
```diff
@@ -749,7 +760,7 @@ def prep_examples(self, num_fewshot: int, prompt_string: str, example_delimiter:
                 for fewshot_idx in fewshot_idxs:
                     context_options, continuation, gold_idx = self.samples[fewshot_idx][
                         'context_options'], self.samples[fewshot_idx]['continuation'], self.samples[fewshot_idx]['gold']
-                    context = context_options[gold_idx]
+                    context = context_options[gold_idx]  # pyright: ignore [reportGeneralTypeIssues]
```
Comment (dakinggg) [inline composer/datasets/in_context_learning_evaluation.py:763]: same

**Review by dakinggg — COMMENTED** [review]
Can you verify that this doesn't change some numbers that we've compute before? like rerun neo 125m eval?

File: composer/datasets/in_context_learning_evaluation.py (hunk 1 above)
Comment (bmosaicml) [reply composer/datasets/in_context_learning_evaluation.py:543]: I think because of the strip_data call `gold_idx` now has an unknown type?

Comment (dakinggg) [reply composer/datasets/in_context_learning_evaluation.py:543]: ok, can you assert that its an int instead of type ignore?

**Review by dakinggg — APPROVED** [review]

**Pull request merged**